tests/test_func.py

```diff
@@ -1,3 +1,5 @@
+from concurrent import futures
+import threading
```

Comment:
Pardon my nitpicking, but this would fail on platforms where threading is disabled. Although `dummy_threading` was [deprecated in Python 3.7](https://docs.python.org/3/library/dummy_threading.html), `cachetools` still supports Python 3.5 and 3.6, so it would be nice if we could skip the test in this (very unlikely) case.